xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingExchange.java

```diff
@@ -18,37 +22,61 @@
 public class KrakenStreamingExchange extends KrakenExchange implements StreamingExchange {
 
     private static final Logger LOG = LoggerFactory.getLogger(KrakenStreamingExchange.class);
-
+    private final static String USE_BETA = "Use_Beta";
     private static final String API_URI = "wss://ws.kraken.com";
+    private static final String API_AUTH_URI = "wss://ws-auth.kraken.com";
+    private static final String API_BETA_URI = "wss://beta-ws.kraken.com";
+//    private static final String API_FUTURES_URI = "wss://futures.kraken.com";
 
-    private final KrakenStreamingService streamingService;
+    private KrakenStreamingService streamingService, privateStreamingService;
+//    private KrakenStreamingService futuresStreamingService;
     private KrakenStreamingMarketDataService streamingMarketDataService;
-    
-    private final SynchronizedValueFactory<Long> nonceFactory = new CurrentTimeNonceFactory();
+    private KrakenStreamingTradeService streamingTradeService;
+
+    private final SynchronizedValueFactory<Long> nonceFactory = new AtomicLongCurrentTimeIncrementalNonceFactory();
 
     public KrakenStreamingExchange() {
-        this.streamingService = new KrakenStreamingService(API_URI);
+    }
+
+    private static String pickUri(boolean isPrivate, boolean useBeta) {
+        return useBeta ? API_BETA_URI : isPrivate ? API_AUTH_URI : API_URI;
     }
 
     @Override
     protected void initServices() {
         super.initServices();
-        streamingMarketDataService = new KrakenStreamingMarketDataService(streamingService);
+        Boolean useBeta = MoreObjects.firstNonNull((Boolean)exchangeSpecification.getExchangeSpecificParametersItem(USE_BETA), Boolean.FALSE);
+
+        this.streamingService = new KrakenStreamingService(false, pickUri(false,useBeta));
+        this.streamingMarketDataService = new KrakenStreamingMarketDataService(streamingService);
+
+        if (StringUtils.isNotEmpty(exchangeSpecification.getApiKey())) {
+            this.privateStreamingService = new KrakenStreamingService(true, pickUri(true,useBeta));
+//            this.futuresStreamingService = new KrakenStreamingService(true, API_FUTURES_URI);
```

Comment:
Best to remove this commented out code. Code related to this is also going to be needed in other places (`connect`/`disconnect` etc) so it's less confusing to leave it out and start a different PR to add futures.

Reply:
done